src/dpg_slam/dpg_slam.cc

```diff
@@ -724,10 +825,60 @@ namespace dpg_slam {
                 }
             }
         }
+	ROS_INFO_STREAM("Active static Map Size: " << active_static_points.size());
+	ROS_INFO_STREAM("Active added Map Size: " << active_added_points.size());
+	ROS_INFO_STREAM("Dynamic added Map Size: " << dynamic_added_points.size());
+	ROS_INFO_STREAM("Dynamic removed Map Size: " << dynamic_removed_points.size());
+    }
+
+    void DpgSLAM::executeDPG(){
+	// Should we do this only once??
+	std::pair<std::vector<occupancyGrid>, occupancyGrid> Grids = computeLocalSubMap();
+	ROS_INFO_STREAM("Occupancy Grids Created");
+	std::vector<occupancyGrid> currentPoseChainGrids = Grids.first;
+	occupancyGrid localSubMapGrid = Grids.second;
+	std::vector<PointIdInfo> removedPoints;
+	detectAndLabelChangesForCurrentPoseChain(currentPoseChainGrids, localSubMapGrid, removedPoints);
+	ROS_INFO_STREAM("Changes are Detected and Updated");
+	updateNodesAndSectorStatus(removedPoints);
+	ROS_INFO_STREAM("Deactivated sectors intersecting with removed points");
+	// Updating map points.
+	std::vector<Vector2f> active_static_points;
+       	std::vector<Vector2f> active_added_points;
+        std::vector<Vector2f> dynamic_removed_points; 
+	std::vector<Vector2f> dynamic_added_points;
+	getActiveAndDynamicMapPoints(active_static_points, active_added_points, dynamic_removed_points, dynamic_added_points);
+	active_static_points_ = active_static_points;
+	active_added_points_ = active_added_points;
+	dynamic_removed_points_ = dynamic_removed_points;
+	dynamic_added_points_ = dynamic_added_points;
```

Comment:
You could just pass these global vars in instead? (If you need to modify getActiveAndDynamicMapPoints to clear the vectors first, that seems fine).

Reply:
That's what I had in mind.